'The "Microsoft.DotNet.Compatibility.ValidatePackage" task failed unexpectedly' in libs legs

Logs from https://dev.azure.com/dnceng/public/_build/results?buildId=1400355&view=logs&jobId=000ba8e3-34f8-51d2-8d06-9b61f10256d0&j=000ba8e3-34f8-51d2-8d06-9b61f10256d0&t=ca9b6e26-8c16-57f9-673e-2bc0dc655f25
```sh
 /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018: The "Microsoft.DotNet.Compatibility.ValidatePackage" task failed unexpectedly. [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018: System.TypeLoadException: Method 'CompareSourceLocations' in type 'Microsoft.CodeAnalysis.CSharp.CSharpCompilation' from assembly 'Microsoft.CodeAnalysis.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation. [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.ApiCompatibility.AssemblySymbolLoader..ctor(Boolean resolveAssemblyReferences) [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.PackageValidation.ApiCompatRunner.GetAssemblySymbolFromStream(Stream assemblyStream, MetadataInformation assemblyInformation, Boolean& resolvedReferences) in /_/src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs:line 111 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.PackageValidation.ApiCompatRunner.RunApiCompat() in /_/src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs:line 49 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.PackageValidation.CompatibleFrameworkInPackageValidator.Validate(Package package) in /_/src/Compatibility/Microsoft.DotNet.PackageValidation/CompatibleFrameworkInPackageValidator.cs:line 62 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.Compatibility.ValidatePackage.ExecuteCore() in /_/src/Compatibility/Microsoft.DotNet.Compatibility/ValidatePackage.cs:line 92 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() in /_/src/Tasks/Common/TaskBase.cs:line 38 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.DotNet.Compatibility.ValidatePackage.Execute() in /_/src/Compatibility/Microsoft.DotNet.Compatibility/ValidatePackage.cs:line 49 [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
  /__w/1/s/artifacts/source-build/self/package-cache/microsoft.dotnet.compatibility/1.0.0-rc.1.21459.41/build/Microsoft.NET.Compatibility.Common.targets(32,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/__w/1/s/artifacts/source-build/self/src/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]
    System.Security.Cryptography.X509Certificates -> /__w/1/s/artifacts/source-build/self/src/artifacts/bin/System.Security.Cryptography.X509Certificates/net7.0-Unix-Release/System.Security.Cryptography.X509Certificates.dll
```

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -242,9 +242,9 @@
       <Uri>https://github.com/dotnet/roslyn-analyzers</Uri>
       <Sha>9cd12bf61357662069562f350e1bf09a87a0940e</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.Compatibility" Version="1.0.0-rc.2.21501.51">
+    <Dependency Name="Microsoft.DotNet.Compatibility" Version="2.0.0-alpha.1.21525.7">
       <Uri>https://github.com/dotnet/sdk</Uri>
-      <Sha>29887967f1f33474b2814d492b3a0dbcf9ce4864</Sha>
+      <Sha>957ae5ca599fdeaee425d23928d42da711373a5e</Sha>
     </Dependency>
   </ToolsetDependencies>
 </Dependencies>

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -242,7 +242,7 @@
       <Uri>https://github.com/dotnet/roslyn-analyzers</Uri>
       <Sha>9cd12bf61357662069562f350e1bf09a87a0940e</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.Compatibility" Version="2.0.0-alpha.1.21525.7">
+    <Dependency Name="Microsoft.DotNet.Compatibility" Version="2.0.0-alpha.1.21525.11">
       <Uri>https://github.com/dotnet/sdk</Uri>
       <Sha>957ae5ca599fdeaee425d23928d42da711373a5e</Sha>
     </Dependency>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -52,10 +52,8 @@
     <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
     <MicrosoftCodeAnalysisCSharpVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>7.0.0-preview1.21521.1</MicrosoftCodeAnalysisNetAnalyzersVersion>
-    <!-- Pin compiler version to workaround issue: https://github.com/dotnet/runtime/issues/59908 -->
-    <MicrosoftNetCompilersToolsetVersion>4.0.0-5.21453.15</MicrosoftNetCompilersToolsetVersion>
     <!-- SDK dependencies -->
-    <MicrosoftDotNetCompatibilityVersion>1.0.0-rc.2.21501.51</MicrosoftDotNetCompatibilityVersion>
+    <MicrosoftDotNetCompatibilityVersion>2.0.0-alpha.1.21525.7</MicrosoftDotNetCompatibilityVersion>
     <!-- Arcade dependencies -->
     <MicrosoftDotNetApiCompatVersion>7.0.0-beta.21521.5</MicrosoftDotNetApiCompatVersion>
     <MicrosoftDotNetBuildTasksFeedVersion>7.0.0-beta.21521.5</MicrosoftDotNetBuildTasksFeedVersion>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -53,7 +53,7 @@
     <MicrosoftCodeAnalysisCSharpVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>7.0.0-preview1.21521.1</MicrosoftCodeAnalysisNetAnalyzersVersion>
     <!-- SDK dependencies -->
-    <MicrosoftDotNetCompatibilityVersion>2.0.0-alpha.1.21525.7</MicrosoftDotNetCompatibilityVersion>
+    <MicrosoftDotNetCompatibilityVersion>2.0.0-alpha.1.21525.11</MicrosoftDotNetCompatibilityVersion>
     <!-- Arcade dependencies -->
     <MicrosoftDotNetApiCompatVersion>7.0.0-beta.21521.5</MicrosoftDotNetApiCompatVersion>
     <MicrosoftDotNetBuildTasksFeedVersion>7.0.0-beta.21521.5</MicrosoftDotNetBuildTasksFeedVersion>

